### PR TITLE
Content Manager - Add `CtiApiDownloader` and `ApiDownloader` documentation

### PR DESCRIPTION
**Pull request by tsarquis88**
|Related issue|
|---|
|#19865 |

## Description

This PR adds two new files that document the way the `CtiApiDownloader` and `ApiDownloader` classes work.

### CODE REVIEW

File: src/shared_modules/content_manager/doc/components/EXECUTION_CONTEXT.md
Comment (tsarquis88) [inline src/shared_modules/content_manager/doc/components/EXECUTION_CONTEXT.md:1]: Changes here are just to normalize all the documentation files.

File: src/shared_modules/content_manager/doc/components/GZIP_DECOMPRESSOR.md
Comment (tsarquis88) [inline src/shared_modules/content_manager/doc/components/GZIP_DECOMPRESSOR.md:1]: Changes here are just to normalize all the documentation files.

File: src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# CTI API Downloader stage
+
+## Details
+
+The [CTI API downloader](../../src/components/CtiApiDownloader.hpp) stage is part of the Content Manager orchestration and is in charge of downloading content from a CTI API to be then processed by the following stages. The downloaded content is stored in one or multiple output files, whose paths will be then published for the consumers to read.
+
+The output content files can be stored into any of the two following output directories:
```
Comment (sebasfalcone) [inline src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md:7]: Nitpick: What about? Enfasis correction
```suggestion
The output content files will be stored in any of the following output directories:
```

File: src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- Downloads folder: If the input files are compressed.
+- Contents folder: If the input files are not compressed.
+
+The download process can be summarized as follows:
+1. Get the last CTI API consumer offset. This is done by performing an HTTP GET query to CTI. This value will be used as the last possible offset to query.
+2. Set the range of offsets to be downloaded, starting from `currentOffset` (set in the context) and with a range-width of `1000`. So, for example, if the current offset is equal to `N`, the range will be from offset `N` to offset `N + 1000`. 
+3. Download the offsets range from **step 2** and update `currentOffset` with the last downloaded offset.
+4. Dump the downloaded offsets into an output file. This file path will be generated as `<output-folder>/<currentOffset>-<contentFileName>`.
+5. Push the new file path (from **step 4**) to the context [data paths](../../src/components/updaterContext.hpp).
+6. If the last possible offset (from **step 1**) has been downloaded, the process finishes. Otherwise, the process continues with the **step 2**.
```
Comment (sebasfalcone) [inline src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md:17]: +1: Really clear

File: src/shared_modules/content_manager/doc/components/API_DOWNLOADER.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+If the download is successful, this stage also updates the context [data paths](../../src/components/updaterContext.hpp) with the download destination path.
+
+## Relation with the UpdaterContext
+
+The context fields related to this stage are:
+
+- `configData`
+  + `url`: Used as the API URL to download from.
+  + `compressionType`: Used to know whether the input file is compressed or not.
```
Comment (sebasfalcone) [inline src/shared_modules/content_manager/doc/components/API_DOWNLOADER.md:19]: Nitpick:
```suggestion
  + `compressionType`: Used to determine whether the input file is compressed or not.
```

File: src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- `/tmp/contents/3000-data.json` (data from offsets 2000 to 3000)
+- `/tmp/contents/3200-data.json` (data from offsets 3000 to 3200)
+
+## Relation with the UpdaterContext
+
+The context fields related to this stage are:
+
+- `configData`
+  + `url`: Used as the CTI API URL to download from.
+  + `compressionType`: Used to know whether the input file is compressed or not.
```
Comment (sebasfalcone) [inline src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md:40]: Nitpick:
```suggestion
  + `compressionType`: Used to determine whether the input file is compressed or not.
```

File: src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  + `contentfileName`: Used as name for the output content file.
+- `downloadsFolder`: Used as output folder when the input file is compressed.
+- `contentsFolder`: Used as output folder when the input file is not compressed.
+- `data`: Used to read and update the paths under the `paths` key. The stage status is also updated on this member.
+- `outputFolder`: Used as the destination file path base.
+- `currentOffset`: Used as the first offset that will be fetched from the API. It is also updated with the last offset fetched, so next time the download begins with the new (and more advanced) offset value.
```
Comment (sebasfalcone) [inline src/shared_modules/content_manager/doc/components/CTI_API_DOWNLOADER.md:46]: Improve: I think we can omit the advanced, its a bit confusing 
```suggestion
- `currentOffset`: Used as the first offset that will be fetched from the API. It is also updated with the last offset fetched, so next time the download begins with the new offset value.
```

**Review by sebasfalcone — COMMENTED** [review]
Really good job

**Review by sebasfalcone — APPROVED** [review]
LGTM!

**Review by jftuduri — APPROVED** [review]
LGTM

**Review by kevin-mcm — APPROVED** [review]

**Review by Damian-Mangold — APPROVED** [review]
LGTM !

**Review by Damian-Mangold — APPROVED** [review]
LGTM !

**Pull request merged**